src/agent/rustjail/src/mount.rs

```diff
@@ -1010,18 +1011,24 @@ lazy_static! {
     };
 }
 
+fn permissions_from_path(path: &Path) -> Result<u32> {
+    match fs::metadata(path) {
+        Ok(metadata) => Ok(metadata.permissions().mode()),
+        Err(e) if e.kind() == ErrorKind::NotFound => Ok(0),
+        Err(e) => Err(e.into()),
+    }
+}
+
 fn mknod_dev(dev: &LinuxDevice, relpath: &Path) -> Result<()> {
     let f = match LINUXDEVICETYPE.get(dev.typ().as_str()) {
         Some(v) => v,
         None => return Err(anyhow!("invalid spec".to_string())),
     };
 
-    let file_mode = dev
-        .file_mode()
-        // drop the mode if it is 0
-        .filter(|&m| m != 0)
-        // fall back to 0o666
-        .unwrap_or(0o666);
+    let file_mode = match dev.file_mode().unwrap_or(0) {
+        0 => permissions_from_path(Path::new(dev.path()))?,
+        x => x,
+    };
```

Comment:
Thanks @zvonkok ❤️. Some minor suggestions, the rest LGTM 

```suggestion
    let file_mode = match dev.file_mode() {
        Some(x) => x,
        None => permissions_from_path(Path::new(dev.path()))?,
    };
```

Reply:
@RuoqingHe  your code  `Some(0) => 0,` I want to check explicilty on `0`, your suggestion would ignore 0 and just return it. I need to make sure if None or 0 get the permissions_from_path and not only in the case of None.